library/spdm_requester_lib/libspdm_req_communication.c

```diff
@@ -137,6 +137,13 @@ return_status libspdm_start_session(IN void *context, IN boolean use_psk,
         DEBUG((DEBUG_INFO,
                "libspdm_start_session - spdm_send_receive_finish - %p\n",
                status));
+        if (spdm_context->last_spdm_error.error_code == SPDM_ERROR_CODE_DECRYPT_ERROR) {
```

Comment:
Had better move it into finish().

Reply:
Yes. I have changed the check to libspdm_receive_response().